deadlines/src/main/java/com/palantir/deadlines/Deadlines.java

```diff
@@ -114,39 +118,95 @@ public static <T> void encodeToRequest(
         long proposedDeadlineNanos = proposedDeadline.toNanos();
         if (stateDeadline == null) {
             // use proposedDeadline
-            checkExpiration(proposedDeadlineNanos, false, false, false);
+            checkExpiration(
+                    proposedDeadlineNanos,
+                    false,
+                    false,
+                    false,
+                    // don't bother with enforcement here
+                    // if we don't have any existing deadline state, then we're merely checking that the caller
+                    // didn't pass in a negative deadline to this method
+                    // eventually, enforcement will be on by default anyway
+                    false);
             adapter.setHeader(
                     request, DeadlinesHttpHeaders.EXPECT_WITHIN, durationToHeaderValue(proposedDeadlineNanos));
+            // we omit the enforcement flag header here to allow the next hop to decide the enforcement strategy
         } else {
             // use the minimum of proposedDeadline and the one read from state
             long remainingStateDeadlineNanos = stateDeadline.remainingNanos(getClockNanoTime());
+            boolean enforced = stateDeadline.enforced();
             if (proposedDeadlineNanos <= remainingStateDeadlineNanos) {
                 boolean proposedDeadlineAlreadyExpired = proposedDeadline.isNegative() || proposedDeadline.isZero();
                 checkExpiration(
                         proposedDeadlineNanos,
                         false,
                         stateDeadline.disablePropagation(),
-                        proposedDeadlineAlreadyExpired);
+                        proposedDeadlineAlreadyExpired,
+                        enforced);
                 if (!stateDeadline.disablePropagation()) {
                     adapter.setHeader(
                             request, DeadlinesHttpHeaders.EXPECT_WITHIN, durationToHeaderValue(proposedDeadlineNanos));
+                    adapter.setHeader(request, DeadlinesHttpHeaders.EXPECT_WITHIN_ENFORCED, Boolean.toString(enforced));
```

Comment:
I think we only want to set the value to `false` here when we've disabled enforcement propagation